### PR TITLE
#1183 Delete Team Endpoint

### LINKED ISSUE #1183: Teams - Delete Team Endpoint
### Description

We want an endpoint that gives us the ability to delete a team if the engineers decide on changing the structure of the teams

### Acceptance Criteria

- [x] New Delete Receiver in the teams routes file
  - Receiver not have a body
  - Route will be /teams/:teamId/delete
- [x] New Controller function called deleteTeam in the teams controller file
  - Will get the current user
  - Will parse the team id out of the request
  - Calls Service function
- [x] New service function called deleteTeam in teams services file
  - Function will throw if user is not admin or above
  - Function will hard delete the team in the database
  - return 204 no content
- [x] Unit Tests

### Proposed Solution

^

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by walker-sean**
## Changes

Endpoint created for hard-deleting teams from the backend

## Screnshots
image
image
image
image

## Test Cases

- Team not found
- Deleter not an admin
- Success
## Checklist

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1183

### CODE REVIEW

**Review by Peyton-McKee — COMMENTED** [review]
what happens to all the projects that have this team as their team? do errors occur or is everything fine?

**Comment by walker-sean** [comment]
> what happens to all the projects that have this team as their team? do errors occur or is everything fine?

The team is removed from their teams field, haven't seen any errors

**Comment by Peyton-McKee** [comment]
> > what happens to all the projects that have this team as their team? do errors occur or is everything fine?
> 
> The team is removed from their teams field, haven't seen any errors

Absolute W

File: src/backend/src/controllers/teams.controllers.ts [hunk 1]
```diff
@@ -62,4 +62,15 @@ export default class TeamsController {
       next(error);
     }
   }
+
+  static async deleteTeam(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { teamId } = req.params;
+      const deleter = await getCurrentUser(res);
+      await TeamsService.deleteTeam(deleter, teamId);
+      return res.status(204);
```
Comment (RChandler234) [inline src/backend/src/controllers/teams.controllers.ts:71]: you should return a message on the response saying 'Successfully deleted team with id ...', or just return the deleted team.

File: src/backend/src/services/teams.services.ts [hunk 2]
```diff
@@ -162,4 +167,13 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  static async deleteTeam(deleter: User, teamId: string): Promise<void> {
```
Comment (RChandler234) [inline src/backend/src/services/teams.services.ts:171]: this needs a comment like all the other service methods

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/TeamsPage/Teams.tsx [hunk 3]
```diff
@@ -6,12 +6,12 @@
...
 import { routes } from '../../utils/routes';
 import TeamsPage from './TeamsPage';
-import TeamSpecificPage from './TeamSpecificPage';
+import TeamDetailPage from './TeamSpecificPage';
 
 const Teams: React.FC = () => {
   return (
     <Switch>
-      <Route path={routes.TEAMS_BY_ID} component={TeamSpecificPage} />
+      <Route path={routes.TEAMS_BY_ID} component={TeamDetailPage} />
```
Comment (RChandler234) [inline src/frontend/src/pages/TeamsPage/Teams.tsx:14]: what's goin on here?

File: src/frontend/src/apis/teams.api.ts [hunk 4]
```diff
@@ -37,6 +37,10 @@ export const setTeamHead = (id: string, userId: number) => {
   });
 };
 
+export const deleteTeam = (id: string) => {
+  return axios.post(apiUrls.teamsDelete(id));
```
Comment (RChandler234) [inline src/frontend/src/apis/teams.api.ts:41]: .post<{message: string}>

File: src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx [hunk 5]
```diff
@@ -40,6 +63,9 @@ const TeamSpecificPage: React.FC = () => {
           <DescriptionPageBlock team={data} />
         </Grid>
       </Grid>
+      {showDeleteModal && (
```
Comment (RChandler234) [inline src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx:66]: don't need showDeleteModal && and the showModal prop

File: src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx [hunk 6]
```diff
@@ -17,12 +23,29 @@ interface ParamTypes {
...
+      onClick={() => setShowDeleteModal(true)}
+    >
+      Delete
+    </NERButton>
+  );
+
   return (
-    <PageLayout title={`Team ${data.teamName}`} previousPages={[{ name: 'Teams', route: routes.TEAMS }]}>
+    <PageLayout
+      headerRight={deleteButton}
```
Comment (RChandler234) [inline src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx:45]: imo, just don't even show the button if someone isn't an admin

File: src/frontend/src/pages/TeamsPage/DeleteTeamModalContainer/DeleteTeamView.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+  if (isLoading || !team) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  const onSubmitWrapper = async (data: DeleteTeamInputs) => {
+    await onSubmit(data);
+  };
+
+  return (
+    <Dialog open={showModal} onClose={onHide}>
```
Comment (RChandler234) [inline src/frontend/src/pages/TeamsPage/DeleteTeamModalContainer/DeleteTeamView.tsx:60]: we have a component for modals now: NERFormModal, if you use that, you should be able to put all the code in just one file: DeleteTeamModal.tsx

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/TeamsPage/Teams.tsx (hunk 3 above)
Comment (walker-sean) [reply src/frontend/src/pages/TeamsPage/Teams.tsx:14]: Oh I was messing with some naming but figured it wasn't worth it - mustve forgotten to change that line back

**Review by RChandler234 — APPROVED** [review]
Amazing

**Pull request merged**